.github/workflows/docker-build-and-push.yml

```diff
@@ -0,0 +1,68 @@
+# Based on https://github.com/docker/build-push-action
+
+name: "Docker Multiarch Build & Push"
+
+on:
+  release:
+    types: [published]
+
+env:
+  DOCKER_HUB_USER: matrixdotorg
+
+jobs:
+  BuildAndPush:
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        name: Checkout
+        uses: actions/checkout@v2
+      - 
+        name: Get release tag
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Login to DockerHub
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ env.DOCKER_HUB_USER }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+      -
+        name: Build temporary (builder) image
```

Comment:
Why are we calling this a temporary builder image?

Reply:
From what I've gathered, that's the image which contains the go toolchain and which builds Dendrite, and only the binaries get copied to the actual monolith and polylith images (to save space I believe - Neil mentioned this somewhere), while this image is just discarded, hence it is temporary and a builder. That's mostly just from a quick glance at the `Dockerfile`s though so I may be wrong.

Reply:
Cool OK, and thanks for the info. Do you want to try that? Otherwise I can do so sometime tomorrow possibly.

Reply:
I've done those tests already on my fork ( I'm nowhere near confident enough to open a PR without testing :P ), and it seemed to work pretty well, but I think it might be good for you to test also, to avoid "works on my end" kinds of bugs :P